src/info.rs

```diff
@@ -422,39 +439,22 @@ impl Info {
         Ok(CommitInfo::new(head_oid, refs_info))
     }
 
-    // Return first n most active commiters as authors within this project.
-    async fn get_authors(dir: &str, no_merges: bool, n: usize) -> Vec<(String, usize, usize)> {
-        let mut args = vec!["-C", dir, "log", "--format='%aN'"];
-        if no_merges {
-            args.push("--no-merges");
-        }
-
-        let output = Command::new("git")
-            .args(args)
-            .output()
-            .await
-            .expect("Failed to execute git.");
-
-        // create map for storing author name as a key and their commit count as value
+    fn get_authors(git_history: &[String], n: usize) -> Vec<(String, usize, usize)> {
         let mut authors = std::collections::HashMap::new();
         let mut total_commits = 0;
-        let output = String::from_utf8_lossy(&output.stdout);
-        for line in output.lines() {
-            let commit_count = authors.entry(line.to_string()).or_insert(0);
+        for line in git_history {
+            let commit_author = line.split('\u{09}').collect::<Vec<_>>()[1].to_string();
+            let commit_count = authors.entry(commit_author.to_string()).or_insert(0);
             *commit_count += 1;
             total_commits += 1;
         }
 
-        // sort authors by commit count where the one with most commit count is first
         let mut authors: Vec<(String, usize)> = authors.into_iter().collect();
         authors.sort_by_key(|(_, c)| *c);
         authors.reverse();
```

Comment:
While we're adjusting these functions, this should probably be simplified.
```suggestion
        authors.sort_by(|(_, a_count), (_, b_count)| b_count.cmp(a_count));
```